server/src/main/java/org/elasticsearch/action/admin/cluster/health/TransportClusterHealthAction.java

```diff
@@ -246,7 +256,9 @@ public void onClusterServiceClose() {
 
                 @Override
                 public void onTimeout(TimeValue timeout) {
-                    sendResponse(task, request, observer.setAndGetObservedState(), waitCount, TimeoutState.TIMED_OUT, listener);
+                    executor.execute(
+                        () -> sendResponse(task, request, observer.setAndGetObservedState(), waitCount, TimeoutState.TIMED_OUT, listener)
+                    );
```

Comment:
I would rather we submitted an `AbstractRunnable` rather than a bare `Runnable` to the executor in all cases - here for instance it'd look like this:

```java
                    executor.execute(
                        ActionRunnable.wrap(
                            listener,
                            l -> sendResponse(task, request, observer.setAndGetObservedState(), waitCount, TimeoutState.TIMED_OUT, l)
                        )
                    );
```

The reason is that [`AbstractRunnable` allows to handle rejections properly](https://github.com/elastic/elasticsearch/blob/fe1083f6c5f29e1fc28b82c1b71be9b7c1ec7a73/server/src/main/java/org/elasticsearch/common/util/concurrent/EsThreadPoolExecutor.java#L74) (and also makes sure we handle other exceptions properly too). Doesn't make a huge difference here today but seems safest to avoid future problems in this area.